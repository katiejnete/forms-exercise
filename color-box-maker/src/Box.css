.box {
    background-color: var(--box-bg-color);
    width: var(--box-width);
    height: var(--box-height);
    border: 1px solid black;
}