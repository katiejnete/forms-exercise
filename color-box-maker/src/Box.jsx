import React from "react";
import "./Box.css";

const Box = ({ id, backgroundColor, width, height, deleteBox }) => {
  return (
    <>
      <div
        className="box"
        data-testid="box"
        style={{
          "--box-bg-color": backgroundColor,
          "--box-width": `${width}px`,
          "--box-height": `${height}px`,
        }}
      ></div>
      <button onClick={() => deleteBox(id)}>X</button>
    </>
  );
};

export default Box;
