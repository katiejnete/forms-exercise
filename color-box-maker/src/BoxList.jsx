import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import { v4 as uuid } from "uuid";

const BoxList = () => {
  const INITIAL_STATE = [];
  const [boxes, setBoxes] = useState(INITIAL_STATE);
  const addBox = (box) => {
    const newBox = { ...box, id: uuid() };
    setBoxes((boxes) => [...boxes, newBox]);
  };
  function deleteBox(id) {
    setBoxes(boxes.filter((box) => box.id !== id));
  }

  return (
    <>
      <NewBoxForm addBox={addBox} />
      {boxes.map(({ id, backgroundColor, width, height }) => (
          <Box
            key={id}
            id={id}
            backgroundColor={backgroundColor}
            width={width}
            height={height}
            deleteBox={deleteBox}
          />
        ))}
    </>
  );
};

export default BoxList;
