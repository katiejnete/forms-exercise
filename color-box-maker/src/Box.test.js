import React from "react";
import {render} from "@testing-library/react";
import Box from "./Box.jsx";
import {deleteBox} from "./BoxList.jsx";

// smoke test
it("renders without crashing", () => {
    render(<Box id="randomId" backgroundColor="black" width={50} height={50} deleteBox={deleteBox}/>);
})

// snapshot test
it("matches snapshot", () => {
    const {asFragment} = render(<Box id="randomId" backgroundColor="black" width={50} height={50} deleteBox={deleteBox}/>);
    expect(asFragment()).toMatchSnapshot();
})