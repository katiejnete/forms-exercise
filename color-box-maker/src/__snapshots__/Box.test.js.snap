// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="box"
    data-testid="box"
    style="--box-bg-color: black; --box-width: 50px; --box-height: 50px;"
  />
  <button>
    X
  </button>
</DocumentFragment>
`;
